src/api/v2/types.rs

```diff
@@ -244,19 +244,19 @@ impl Reply for SubscriptionId {
 }
 
 #[derive(Deserialize)]
+#[serde(tag = "type")]
 #[serde(rename_all = "kebab-case")]
 pub enum RequestType {
 	Version,
 	Status,
-	Submit,
+	Submit { message: Transaction },
 }
 
 #[derive(Deserialize)]
 pub struct Request {
-	#[serde(rename = "type")]
+	#[serde(flatten)]
 	pub request_type: RequestType,
```

Comment:
I think this is no longer request type, but rather request itself. Can you rename it?